### PR TITLE
Added Continued fractions

### PR DESCRIPTION
**Pull request by Saksham1970**
### Describe your change:
added a function to convert a rational number to continued fraction list

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/continued_fraction.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    [0, 2, 4]
+    >>> continued_fraction(Fraction("415/93"))
+    [4, 2, 6, 7]
+    """
+    numerator, denominator = num.as_integer_ratio()
+    continued_fraction_list: list[int] = []
+    while True:
+        integer_part = int(numerator / denominator)
+        continued_fraction_list.append(integer_part)
+        numerator = numerator - integer_part * denominator
```
Comment (CaedenPH) [inline maths/continued_fraction.py:38]:
```suggestion
        numerator -= integer_part * denominator
```

File: maths/continued_fraction.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        if numerator == 0:
+            break
+        numerator, denominator = denominator, numerator
+
+    return continued_fraction_list
+
+
+if __name__ == "__main__":
+
```
Comment (CaedenPH) [inline maths/continued_fraction.py:47]:
```suggestion
```

**Review by CaedenPH — COMMENTED** [review]
@cclauss Looks good - 
Small changes

**Review by tianyizheng02 — APPROVED** [review]

File: maths/continued_fraction.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    >>> continued_fraction(Fraction(2))
+    [2]
+    >>> continued_fraction(Fraction("3.245"))
+    [3, 4, 12, 4]
+    >>> continued_fraction(Fraction("2.25"))
+    [2, 4]
+    >>> continued_fraction(1/Fraction("2.25"))
+    [0, 2, 4]
+    >>> continued_fraction(Fraction("415/93"))
+    [4, 2, 6, 7]
```
[the remark this answers is not in the document]
Comment (tianyizheng02) [reply maths/continued_fraction.py:31]: Added tests for 0 and negative numbers in #8985

**Pull request merged**